nixos/modules/services/databases/mysql.nix

```diff
@@ -404,7 +401,7 @@ in
                   done
                 ''}
 
-                if [ -f /tmp/mysql_init ]
+                if [ -f ${cfg.dataDir}/mysql_init ]
```

Comment:
With `PrivateTmp` is this needed? I would just leave it as `/tmp`.

See comment below about `ReadWritePaths`.

Reply:
With `/tmp/mysql_init` not worked. Folder `/tmp` moved to `/tmp/systemd-private-...-mysql.service-.../tmp`

Reply:
I'll ask the same question here I asked below then: what option is causing the service to have no access to `/tmp` (where `/tmp` is actually a `systemd` managed private `/tmp`)?

Reply:
When initialization a new database, the mysql service restarted. Systemd deletes folder `/tmp/systemd-private-...-mysql.service-...1/tmp` and created new folder `/tmp/systemd-private-...-mysql.service-...2/tmp` without file `mysql_init`.
Error reading `/tmp/mysql_init` and the new database is not initialized

Reply:
This doesn't make sense to me. The `mysql_init` file is created in the `ExecStartPre` script and then removed in the `ExecStartPost` script. Is the `systemd` managed `/tmp` directory changed between `ExecStartPre` and `ExecStartPost`?

Reply:
First start
```
июн 05 18:45:59 NixOS-Test mysql-pre-start[7403]: OK
июн 05 18:45:59 NixOS-Test mysql-pre-start[7403]: To start mysqld at boot time you have to copy
июн 05 18:45:59 NixOS-Test mysql-pre-start[7403]: support-files/mysql.server to the right place for your system
июн 05 18:45:59 NixOS-Test mysql-pre-start[7403]: Two all-privilege accounts were created.
июн 05 18:45:59 NixOS-Test mysql-pre-start[7403]: One is root@localhost, it has no password, but you need to
июн 05 18:45:59 NixOS-Test mysql-pre-start[7403]: be system 'root' user to connect. Use, for example, sudo mysql
июн 05 18:45:59 NixOS-Test mysql-pre-start[7403]: The second is mysql@localhost, it has no password either, but
июн 05 18:45:59 NixOS-Test mysql-pre-start[7403]: you need to be the system 'mysql' user to connect.
июн 05 18:45:59 NixOS-Test mysql-pre-start[7403]: After connecting you can set the password, if you would need to be
июн 05 18:45:59 NixOS-Test mysql-pre-start[7403]: able to connect as any of these users with a password and without sudo
июн 05 18:45:59 NixOS-Test mysql-pre-start[7403]: See the MariaDB Knowledgebase at http://mariadb.com/kb or the
июн 05 18:45:59 NixOS-Test mysql-pre-start[7403]: MySQL manual for more instructions.
июн 05 18:45:59 NixOS-Test mysql-pre-start[7403]: You can start the MariaDB daemon with:
июн 05 18:45:59 NixOS-Test mysql-pre-start[7403]: cd '/nix/store/ls5cb9ha9rh7c9bqy7fdlh9s41qarysr-mariadb-server-10.4.13' ; /nix/store/ls5cb9ha9rh7c9bqy7fdlh9s41qarysr-mariadb-server-10.4.13/bin/mysqld_safe --datadir='/var/data/db/mysql'
июн 05 18:45:59 NixOS-Test mysql-pre-start[7403]: You can test the MariaDB daemon with mysql-test-run.pl
июн 05 18:45:59 NixOS-Test mysql-pre-start[7403]: cd '/nix/store/ls5cb9ha9rh7c9bqy7fdlh9s41qarysr-mariadb-server-10.4.13/mysql-test' ; perl mysql-test-run.pl
июн 05 18:45:59 NixOS-Test mysql-pre-start[7403]: Please report any problems at http://mariadb.org/jira
июн 05 18:45:59 NixOS-Test mysql-pre-start[7403]: The latest information about MariaDB is available at http://mariadb.org/.
июн 05 18:45:59 NixOS-Test mysql-pre-start[7403]: You can find additional information about the MySQL part at:
июн 05 18:45:59 NixOS-Test mysql-pre-start[7403]: http://dev.mysql.com
июн 05 18:45:59 NixOS-Test mysql-pre-start[7403]: Consider joining MariaDB's strong and vibrant community:
июн 05 18:45:59 NixOS-Test mysql-pre-start[7403]: https://mariadb.org/get-involved/
июн 05 18:45:59 NixOS-Test wzckfiawca7npk3q4srhlcg8aib4gyh4-mysql-setup[7479]: MySQL daemon not yet started. Waiting for 1 second...
```
Folder `/tmp/systemd-private-...-mysql.service-...1/tmp` removed
Next start
```
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Note] /nix/store/ls5cb9ha9rh7c9bqy7fdlh9s41qarysr-mariadb-server-10.4.13/bin/mysqld (mysqld 10.4.13-MariaDB) starting as process 7478 ...
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Warning] WSREP: Failed to guess base node address. Set it explicitly via wsrep_node_address.
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Warning] WSREP: Failed to guess base node address. Set it explicitly via wsrep_node_address.
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Warning] WSREP: Guessing address for incoming client connections failed. Try setting wsrep_node_incoming_address explicitly.
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Note] WSREP: Node addr:
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Note] InnoDB: Using Linux native AIO
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Note] InnoDB: Uses event mutexes
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Note] InnoDB: Number of pools: 1
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Note] InnoDB: Using generic crc32 instructions
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Note] mysqld: O_TMPFILE is not supported on /tmp (disabling future attempts)
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [ERROR] InnoDB: Linux Native AIO interface is not supported on this platform. Please check your OS documentation and install appropriate binary of InnoDB.
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Note] InnoDB: You can disable Linux Native AIO by setting innodb_use_native_aio = 0 in my.cnf
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Warning] InnoDB: Linux Native AIO disabled.
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Note] InnoDB: Initializing buffer pool, total size = 128M, instances = 1, chunk size = 128M
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Note] InnoDB: Completed initialization of buffer pool
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Note] InnoDB: 128 out of 128 rollback segments are active.
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Note] InnoDB: Creating shared tablespace for temporary tables
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Note] InnoDB: 10.4.13 started; log sequence number 60981; transaction id 21
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Note] Plugin 'FEEDBACK' is disabled.
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Note] InnoDB: Loading buffer pool(s) from /var/data/db/mysql/ib_buffer_pool
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Note] InnoDB: Buffer pool(s) load completed at 200605 18:45:59
июн 05 18:45:59 NixOS-Test mysqld[7478]: 2020-06-05 18:45:59 0 [Note] Server socket created on IP: '::'.
июн 05 18:46:00 NixOS-Test mysqld[7478]: 2020-06-05 18:46:00 0 [Note] Reading of all Master_info entries succeeded
июн 05 18:46:00 NixOS-Test mysqld[7478]: 2020-06-05 18:46:00 0 [Note] Added new Master_info '' to hash table
июн 05 18:46:00 NixOS-Test mysqld[7478]: 2020-06-05 18:46:00 0 [Note] /nix/store/ls5cb9ha9rh7c9bqy7fdlh9s41qarysr-mariadb-server-10.4.13/bin/mysqld: ready for connections.
июн 05 18:46:00 NixOS-Test mysqld[7478]: Version: '10.4.13-MariaDB'  socket: '/run/mysqld/mysqld.sock'  port: 3306  MariaDB Server
июн 05 18:46:00 NixOS-Test wzckfiawca7npk3q4srhlcg8aib4gyh4-mysql-setup[7479]: Creating initial database: empty_testdb
июн 05 18:46:01 NixOS-Test systemd[1]: Started MySQL Server.
```